lib/src/reporting_query_parser.dart

```diff
@@ -0,0 +1,42 @@
+import 'dart:convert';
+
+class ReportingQueryParser {
+  static const _emptyEntry = '';
+
+  /// Parse queryId from a json string.
+  static String parseQueryIdFromJsonString(String jsonString) =>
+      json.decode(jsonString)['queryId'] ?? _emptyEntry;
+
+  /// Parse google storage download from a json string.
+  static String parseDownloadPathFromJsonString(String jsonString) =>
+      (json.decode(jsonString)['metadata'] ??
+          const {})['googleCloudStoragePathForLatestReport'] ??
+      _emptyEntry;
+
+  /// Parse revenue from a json string.
+  static Map<String, String> parseRevenueFromString(String responseFull) {
+    String responseBody =
+        ((json.decode(responseFull)['gapiRequest'] ?? const {})['data'] ??
+                const {})['body'] ??
+            '';
+
+    if (responseBody.isEmpty) return <String, String>{};
+
+    // Extracts the substring after 'Revenue (USD)' and before 'Report Time',
```

Comment:
Here is a sample output:

Insertion Order ID,Revenue (USD)
8127549,855.952028

Report Time:,2020/07/09 01:14 GMT
Date Range:,2020/06/22 to 2020/07/21
Group By:,Insertion Order ID
MRC Accredited Metrics,Active View metrics are accredited only when Measurement Source = Measured
"Reporting numbers from the previous month are finalized (for billing purposes) by the third of each month, unless communicated otherwise."
Filter by Advertiser ID:,164337
Filter by Insertion Order ID:,8127549